service/controller/chart/v1/resource/status/resource.go

```diff
@@ -62,6 +63,9 @@ func equals(a, b v1alpha1.ChartStatus) bool {
 	if a.Release.LastDeployed != b.Release.LastDeployed {
 		return false
 	}
+	if a.Reason != b.Reason {
```

Comment:
There are cases chart failed multiple times, so we need to compare the `reason` as well.